backpack/core/derivatives/rnn.py

```diff
@@ -0,0 +1,236 @@
+from torch.nn import RNN
+
+from backpack.core.derivatives.basederivatives import BaseParameterDerivatives
+
+
+class RNNDerivatives(BaseParameterDerivatives):
+    """Partial derivatives for the torch.nn.RNN layer.
+
+    a_t = W_ih x_t + b_ih + W_hh h_{t-1} + b_hh
+    h_t = tanh(a_t)
+
+    Index conventions:
+    ------------------
+    * t: Sequence dimension
+    * v: Free dimension
+    * n: Batch dimension
+    * h: Output dimension
+    * i: Input dimension
+    """
+
+    @staticmethod
+    def _check_parameters(module: RNN) -> None:
+        """Check the parameters of module.
+
+        Args:
+            module: module which to check
+
+        Raises:
+            NotImplementedError: If any parameter of module does not match expectation
+        """
+        if module.num_layers > 1:
+            raise NotImplementedError("only num_layers = 1 is supported")
+        if not module.nonlinearity == "tanh":
+            raise NotImplementedError("only nonlinearity = tanh is supported")
+        if module.bias is not True:
+            raise NotImplementedError("only bias = True is supported")
+        if module.batch_first is not False:
+            raise NotImplementedError("only batch_first = False is supported")
+        if not module.dropout == 0:
+            raise NotImplementedError("only dropout = 0 is supported")
+        if module.bidirectional is not False:
+            raise NotImplementedError("only bidirectional = False is supported")
+
+    @staticmethod
+    def _a_jac_t_mat_prod(
+        output: Tensor,
+        weight_hh_l0: Tensor,
+        mat: Tensor,
+    ) -> Tensor:
+        """Calculates jacobian vector product wrt a.
+
+        Args:
+            output: the values of the hidden layer
+            weight_hh_l0: weight matrix hidden-to-hidden
+            mat: matrix to multiply
+
+        Returns:
+            jacobian vector product wrt a
+        """
+        V: int = mat.shape[0]
+        N: int = mat.shape[2]
+        T: int = mat.shape[1]
+        H: int = mat.shape[3]
+        a_jac_t_mat_prod: Tensor = zeros(V, T, N, H, device=mat.device)
+        for t in range(T)[::-1]:
+            if t == (T - 1):
+                a_jac_t_mat_prod[:, t, ...] = einsum(
+                    "vnh,nh->vnh",
+                    mat[:, t, ...],
+                    1 - output[t, ...] ** 2,
+                )
+            else:
+                a_jac_t_mat_prod[:, t, ...] = einsum(
+                    "vnh,nh->vnh",
+                    mat[:, t, ...]
+                    + einsum(
+                        "vng,gh->vnh",
+                        a_jac_t_mat_prod[:, t + 1, ...],
+                        weight_hh_l0,
+                    ),
+                    1 - output[t, ...] ** 2,
+                )
+        return a_jac_t_mat_prod
+
+    def _jac_t_mat_prod(
+        self, module: RNN, g_inp: Tuple[Tensor], g_out: Tuple[Tensor], mat: Tensor
+    ) -> Tensor:
+        return torch.einsum(
```

Comment:
@schaefertim Should there be a `self.check_parameters()` here?

Reply:
Yes.